roles/helm_common/library/helm_cli.py

```diff
@@ -0,0 +1,235 @@
+import os
+import json
+
+from contextlib import contextmanager
+from tempfile import NamedTemporaryFile
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+class Helm(AnsibleModule):
+
+    def __init__(self, *args, **kwargs):
+        AnsibleModule.__init__(
+            self,
+            argument_spec=dict(
+                chart=dict(type='dict'),
+                release=dict(type='str', aliases=['name']),
+                namespace=dict(type='str'),
+                binary=dict(type='str'),
+                state=dict(default='present',
+                           choice=['latest', 'present', 'absent', 'purged']),
+                values=dict(type='list'),
+            )
+        )
+        self._helm_bin = None
+        self.delete_temp_file = True
+
+    def get_chart(self):
+        chart = self.params.get('chart')
+        if chart is None:
+            raise HelmError("'chart' is a mandatory argument for "
+                            "state=present|latest")
+        if 'name' not in chart:
+            raise HelmError("'chart' argument required a 'name' key for "
+                            "state=present|latest")
+        return chart
+
+    @property
+    def helm_bin(self):
+        if self._helm_bin is None:
+            binary = self.params.get('binary')
+            if binary and os.path.isfile(binary):
+                self._helm_bin = binary
+            else:
+                self._helm_bin = self.get_bin_path(
```

Comment:
I still think supporting `binary` to be a path to a directory rather than the expected binary doesn't make sense :) No need to be too 'clever' and work-around bugs written by a consumer of this module.